Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -0,0 +1,658 @@
+            'street_address3' => null,
+            'street_address4' => null,
+        ];
+
+        $shortAddressApplePay = [
+            'country_id' => 'US',
+            'postcode' => 10001,
+            'region' => 'New York',
+            'city' => 'New York'
+        ];
+
+        $this->cartHelper->expects($this->at(0))
+            ->method('getRoundAmount')
+            ->with('5')
+            ->will($this->returnValue((int)500));
+        $this->cartHelper->expects($this->at(1))
+            ->method('getRoundAmount')
+            ->with(0)
+            ->will($this->returnValue(0));
+
+        $shippingAddress = $this->getMockBuilder(\Magento\Quote\Model\Quote\Address::class)
+            ->setMethods(['addData', 'setCollectShippingRates', 'setShippingMethod', 'getGroupedAllShippingRates',
+                'getShippingDiscountAmount', 'getShippingAmount', 'save'
+            ])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $shippingAddress->method('addData')
+            ->with($shortAddressApplePay)
+            ->willReturnSelf();
+        $shippingAddress->method('setShippingMethod')
+            ->withAnyParameters()
+            ->willReturnSelf();
+        $shippingAddress->method('save')
+            ->willReturnSelf();
+        $shippingAddress->expects($this->once())
+            ->method('setCollectShippingRates')
+            ->with(true)
+            ->willReturnSelf();
+        $shippingAddress->method('getShippingDiscountAmount')
+            ->willReturn('0');
+        $shippingAddress->method('getShippingAmount')
+            ->willReturn('5');
+
+        $addressRate = $this->getMockBuilder(\Magento\Quote\Model\Quote\Address\Rate::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $shippingRates =[['flatrate' => $addressRate]];
+        $shippingAddress->expects($this->once())
+            ->method('getGroupedAllShippingRates')
+            ->willReturn($shippingRates);
+
+        $quote = $this->getQuoteMock($shippingAddress);
+
+        $testClass = new \Bolt\Boltpay\Model\Api\ShippingMethods(
+            $this->hookHelper,
+            $this->regionModel,
+            $this->factoryShippingOptionsMock,
+            $this->shippingTaxInterfaceFactory,
+            $this->cartHelper,
+            $this->totalsCollector,
+            $this->converter,
+            $this->shippingOptionInterfaceFactory,
+            $this->bugsnag,
+            $this->logHelper,
+            $this->errorResponse,
+            $this->response,
+            $this->configHelper,
+            $this->request,
+            $this->cache,
+            $this->priceHelper,
+            $this->sessionHelper
+        );
+
+        $result = $testClass->shippingEstimation($quote, $shippingAddressData);
+
+        $this->assertEquals($this->factoryShippingOptionsMock, $result);
+        $this->assertEquals($this->shippingTaxInterfaceFactory, $result->getTaxResult());
```

Comment:
why you need to assert `$result->getTaxResult()` when you already assert `$result` above?

Reply:
removed.